pkg/pgmodel/ingestor/trace/trace_dispatcher.go

```diff
@@ -0,0 +1,58 @@
+package trace
+
+import (
+	"context"
+
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/tracer"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// Manages trace batcher and ingestion
+type Dispatcher struct {
+	batcher *Batcher
+	async   bool // 'true' means that we don't wait for request to be persisted
+}
+
+func NewDispatcher(writer Writer, async bool, batchConfig BatcherConfig) *Dispatcher {
+	batcher := NewBatcher(batchConfig, writer)
+	batcher.Run()
+	return &Dispatcher{batcher: batcher, async: async}
+}
+
+type InsertTracesReq struct {
```

Comment:
Are we using this guy out of `trace` package? Wondering if we need this public or not.

Reply:
Yeah we can make it private